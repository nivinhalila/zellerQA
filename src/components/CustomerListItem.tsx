import React from "react";
import styled from "styled-components";

import { Text } from "./Text";

const Column = styled.div`
  display: flex;
  flex-direction: column;
  align-content: center;
`;

const Initial = styled.div`
  display: flex;
  align-items: center;
  background-color: #e8f2fb;
  border-radius: 0.5rem;
  margin: 0 1.5rem 0 0;
  width: 3.5rem;
  height: 3.5rem;
`;

const InitialText = styled(Text)`
  color: #005cc8;
  flex-grow: 1;
  margin: 0;
  text-align: center;
`;

const NameText = styled(Text)`
  flex-grow: 1;
  margin: 0;
`;

const RoleText = styled(Text)`
  color: #787775;
  font-size: 1.2em;
  flex-grow: 1;
  margin: 0;
`;

const Row = styled.div`
  display: flex;
  flex-direction: "row";
  margin-bottom: 1.5rem;

  &:last-child {
    margin-bottom: 0;
  }
`;

interface CustomerListItemProps {
  key: string;
  name: string;
  role: string;
}

export function CustomerListItem(props: CustomerListItemProps) {
  const { name, role } = props;

  const initial = name.charAt(0).toUpperCase();

  return (
    <Row>
      <Initial>
        <InitialText>{initial}</InitialText>
      </Initial>
      <Column>
        <NameText>{name}</NameText>
        <RoleText>{role}</RoleText>
      </Column>
    </Row>
  );
}

export default CustomerListItem;
