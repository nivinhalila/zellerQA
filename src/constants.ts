import { ZellerAdmin, ZellerManager } from "./types";

export const ZELLER_ADMIN: ZellerAdmin = "ADMIN";
export const ZELLER_MANAGER: ZellerManager = "MANAGER";
