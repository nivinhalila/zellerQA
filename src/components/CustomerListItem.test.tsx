import "jest-styled-components";
import { render } from "@testing-library/react";

import CustomerListItem from "./CustomerListItem";
import { ZELLER_ADMIN } from "../constants";

test("renders customer list item component", () => {
  expect.assertions(4);

  const name = "Jonathan Warykowski";
  const role = ZELLER_ADMIN;

  const { container, getByText } = render(
    <CustomerListItem key="customer1" name={name} role={role} />
  );

  const customerInitial = getByText(name.charAt(0).toUpperCase());
  const customerName = getByText(name);
  const customerRole = getByText(role);

  expect(customerInitial).toBeInTheDocument();
  expect(customerName).toBeInTheDocument();
  expect(customerRole).toBeInTheDocument();

  expect(container.firstChild).toMatchSnapshot();
});
