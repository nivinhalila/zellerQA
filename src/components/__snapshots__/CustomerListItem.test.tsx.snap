// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders customer list item component 1`] = `
.c2 {
  color: #212322;
  font-size: 1.5em;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #e8f2fb;
  border-radius: 0.5rem;
  margin: 0 1.5rem 0 0;
  width: 3.5rem;
  height: 3.5rem;
}

.c3 {
  color: #005cc8;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  margin: 0;
  text-align: center;
}

.c5 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  margin: 0;
}

.c6 {
  color: #787775;
  font-size: 1.2em;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  margin: 0;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: "row";
  -ms-flex-direction: "row";
  flex-direction: "row";
  margin-bottom: 1.5rem;
}

.c0:last-child {
  margin-bottom: 0;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <p
      class="c2 c3"
    >
      J
    </p>
  </div>
  <div
    class="c4"
  >
    <p
      class="c2 c5"
    >
      Jonathan Warykowski
    </p>
    <p
      class="c2 c6"
    >
      ADMIN
    </p>
  </div>
</div>
`;
